torchmetrics/functional/image/ms_ssim.py

```diff
@@ -0,0 +1,191 @@
+    sigma: Sequence[float] = (1.5, 1.5),
+    reduction: str = "elementwise_mean",
+    data_range: Optional[float] = None,
+    k1: float = 0.01,
+    k2: float = 0.03,
+    betas: Union[Tuple[float, float, float, float, float], Tuple[float, ...]] = (
+        0.0448,
+        0.2856,
+        0.3001,
+        0.2363,
+        0.1333,
+    ),
+    normalize: Optional[Literal["relu", "simple"]] = None,
+) -> Tensor:
+    """Computes Multi-Scale Structual Similarity Index Measure. Adapted from: https://github.com/jorge-
+    pessoa/pytorch-msssim/blob/master/pytorch_msssim/__init__.py.
+
+    Args:
+        preds: estimated image
+        target: ground truth image
+        kernel_size: size of the gaussian kernel (default: (11, 11))
+        sigma: Standard deviation of the gaussian kernel (default: (1.5, 1.5))
+        reduction: a method to reduce metric score over labels.
+
+            - ``'elementwise_mean'``: takes the mean (default)
+            - ``'sum'``: takes the sum
+            - ``'none'``: no reduction will be applied
+
+        data_range: Range of the image. If ``None``, it is determined from the image (max - min)
+        k1: Parameter of SSIM.
+        k2: Parameter of SSIM.
+    """
+    sim_list: List[Tensor] = []
+    cs_list: List[Tensor] = []
+
+    if preds.size()[-1] < 2 ** len(betas) or preds.size()[-2] < 2 ** len(betas):
+        raise ValueError(
+            f"For a given number of `betas` parameters {len(betas)}, the image height and width dimensions must be "
+            f"larger than or equal to {2 ** len(betas)}."
+        )
+
+    _betas_div = max(1, (len(betas) - 1)) ** 2
+    if preds.size()[-2] // _betas_div <= kernel_size[0] - 1:
+        raise ValueError(
+            f"For a given number of `betas` parameters {len(betas)} and kernel size {kernel_size[0]}, the image height "
+            f"must be larger than {(kernel_size[0] - 1) * _betas_div}."
+        )
+    if preds.size()[-1] // _betas_div <= kernel_size[1] - 1:
+        raise ValueError(
+            f"For a given number of `betas` parameters {len(betas)} and kernel size {kernel_size[1]}, the image width "
+            f"must be larger than {(kernel_size[1] - 1) * _betas_div}."
+        )
+
+    for _ in range(len(betas)):
+        sim, contrast_sensitivity = _get_normalized_sim_and_cs(
+            preds, target, kernel_size, sigma, reduction, data_range, k1, k2, normalize
+        )
+        sim_list.append(sim)
+        cs_list.append(contrast_sensitivity)
+        preds = F.avg_pool2d(preds, (2, 2))
+        target = F.avg_pool2d(target, (2, 2))
+
+    sim_stack = torch.stack(sim_list)
+    cs_stack = torch.stack(cs_list)
+
+    if normalize == "simple":
+        sim_stack = (sim_stack + 1) / 2
+        cs_stack = (cs_stack + 1) / 2
+
+    sim_stack = sim_stack ** torch.tensor(betas)
+    cs_stack = cs_stack ** torch.tensor(betas)
+    return torch.prod(cs_stack[:-1]) * sim_stack[-1]
+
+
+def ms_ssim(
```

Comment:
There is a minor issue with the docs. Check that all the whitespace (indentation, new lines etc) are correct so that the rendered output is good. You can generate the docs by 

```
cd docs
make html
```
Then open the index.html in the build folder and navigate to the ssim page.

![image](https://user-images.githubusercontent.com/5495193/146679838-f5050997-9f33-404d-b5f7-acd34991fc1e.png)

Reply:
Hopefully should be fixed now. However, I can see changes only when I open `docs/build/html/index.html` and not when I open `docs/build/index.html` - is that okay?.

Reply:
yes, you find it in `docs/build/html/index.html` and to keep it simple you can just run `make docs` in the repo root